### PR TITLE
#362 - Add change() property callback that only gets fired when the property changes.

### PR DESCRIPTION
**Pull request by treshugart**
Implements #362.

### CODE REVIEW

File: test/unit/lifecycle/property.js [hunk 1]
```diff
@@ -323,5 +323,107 @@ describe('lifecycle/property', function () {
...
+          change () {
+            ++calls;
+          }
+        });
+        expect(calls).to.equal(0);
+      });
+
+      it('is called if the property has no default value and it is set', function () {
+        let calls = 0;
+        let elem = create({
```
Comment (bqx) [inline test/unit/lifecycle/property.js]: We could use `const` for declarations like this where we don't need to assign again

File: src/lifecycle/property.js [hunk 2]
```diff
@@ -125,6 +125,10 @@ function createNativePropertyDefinition (name, opts) {
       opts.set(this, changeData);
     }
 
+    if (opts.change && oldValue !== newValue) {
```
Comment (joscha) [inline src/lifecycle/property.js]: `typeof opts.change === 'function'` is more sane

File: test/unit/lifecycle/property.js [hunk 3]
```diff
@@ -323,5 +323,107 @@ describe('lifecycle/property', function () {
         create(opts);
       });
     });
+
+    describe('change()', function () {
+      it('is not called if the property has no default value', function () {
+        let calls = 0;
```
Comment (joscha) [inline test/unit/lifecycle/property.js]: this looks like a pure [sinonJS](http://sinonjs.org/) use-case to me.

File: test/unit/lifecycle/property.js [hunk 4]
```diff
@@ -323,5 +323,107 @@ describe('lifecycle/property', function () {
...
+      it('is not called if the property value does not change', function () {
+        let calls = 0;
+        let elem = create({
+          default: true,
+          change () {
+            ++calls;
+          }
+        });
+        elem.test = true;
+        elem.test = true;
```
Comment (joscha) [inline test/unit/lifecycle/property.js]: copy & paste glitch?

Comment (treshugart) [inline test/unit/lifecycle/property.js]: That's to test whether or not `change()` is called if the value being set again is the same.

Comment (joscha) [inline test/unit/lifecycle/property.js]: O wow - observer power :)

**Pull request merged**